rust_snuba/src/processors/replays.rs

```diff
@@ -0,0 +1,593 @@
+use anyhow::Context;
+use chrono::{DateTime, Utc};
+use md5;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use uuid::Uuid;
+
+use crate::types::{InsertBatch, KafkaMessageMetadata, RowData};
+
+pub fn process_message(
+    payload: KafkaPayload,
+    metadata: KafkaMessageMetadata,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+
+    let replay_message: ReplayMessage = serde_json::from_slice(payload_bytes)?;
+    let replay_payload: ReplayPayload = serde_json::from_slice(&replay_message.payload)?;
+
+    let output = match replay_payload {
+        ReplayPayload::ReplayClickEvent(event) => {
+            let replay_rows = event
+                .clicks
+                .into_iter()
+                .map(|click| ReplayRow {
+                    click_alt: click.alt,
+                    click_aria_label: click.aria_label,
+                    click_class: click.class,
+                    click_id: click.id,
+                    click_is_dead: click.is_dead,
+                    click_is_rage: click.is_rage,
+                    click_node_id: click.node_id,
+                    click_role: click.role,
+                    click_tag: click.tag,
+                    click_testid: click.testid,
+                    click_text: click.text,
+                    click_title: click.title,
+                    event_hash: click.event_hash,
+                    offset: metadata.offset,
+                    partition: metadata.partition,
+                    project_id: replay_message.project_id,
+                    replay_id: replay_message.replay_id,
+                    retention_days: replay_message.retention_days,
+                    timestamp: click.timestamp,
+                    ..Default::default()
+                })
+                .collect::<Vec<ReplayRow>>();
+
+            vec![serde_json::to_vec(&replay_rows)?]
```

Comment:
You don't need a `Vec` here, an array works as well because it's the same size in every branch.
```suggestion
            [serde_json::to_vec(&replay_rows)?]
```

Reply:
I think the first case can return more rows than the others, so it's not going to be the same size.

Reply:
Yes this was obscured due to an oversight on my part.  Marking as resolved 👍